File: backend/app/models/task.py
import uuid
from datetime import datetime
from sqlalchemy.orm import relationship

from app.db.base import Base
from app.db.guid import GUID

class Task(Base):
    __tablename__ = "tasks"

    id = Column(GUID(), primary_key=True, default=uuid.uuid4, index=True)
    title = Column(String, index=True, nullable=False)
    description = Column(String, nullable=True)
    priority = Column(String, nullable=False) # e.g. low, medium, high
    status = Column(String, default="pending", nullable=False, index=True) # pending, in_progress, completed
    deadline = Column(DateTime, nullable=True)
    user_id = Column(GUID(), ForeignKey("users.id"), nullable=False, index=True)
    category_id = Column(GUID(), ForeignKey("categories.id"), nullable=True, index=True)
    created_at = Column(DateTime, default=datetime.utcnow)

    owner = relationship("User", backref="tasks")
    category = relationship("Category", backref="tasks")

File: frontend/src/types/task.ts
export type TaskPriority = "low" | "medium" | "high";
export type TaskStatus = "pending" | "in_progress" | "completed";

export interface Category {
  id: string;
  name: string;
  user_id: string;
}

export interface TaskShare {
  id: string;
  task_id: string;
  user_id: string;
  permission: "read" | "write";
  shared_at: string;
}

export interface Attachment {
  id: string;
  task_id: string;
  file_name: string;
  file_path: string;
  file_type: string;
  file_size: number;
  uploaded_at: string;
}

export interface Task {
  id: string;
  title: string;
  description?: string;
  priority: TaskPriority;
  status: TaskStatus;
  deadline?: string;
  user_id: string;
  category_id?: string;
  created_at: string;
  category?: Category;
  shares?: TaskShare[];
  attachments?: Attachment[];
}

export type TaskCreate = Omit<Task, "id" | "created_at" | "category" | "user_id">;
export type TaskUpdate = Partial<TaskCreate>;
